src/main/java/Duke.java

```diff
@@ -1,10 +1,106 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class Duke {
+    private static Storage storage;
+    private static TaskList tasks;
+
+    /**
+     * Creates a new instance of Duke which stores information at the given path.
+     * @param filePath the path where task information is stored.
+     */
+    public Duke(String filePath) {
+        storage = new Storage(filePath);
+        tasks = new TaskList(storage.read());
+    }
+
+    /**
+     * Contains most of the chat bot logic
+     */
+    public void run() {
```

Comment:
Perhaps extract out certain parts into helper methods? Currently this method seems too long and will only grow longer as more logic is added.